src/junction/markdown/checklists.py

```diff
@@ -0,0 +1,65 @@
+import re
+from typing import List, Any
+
+from markdown import Markdown
+from markdown.extensions import Extension
+from markdown.postprocessors import Postprocessor
+
+
+class ChecklistExtension(Extension):
+    def extendMarkdown(self, md: Markdown) -> None:
+        postprocessor = ChecklistPostprocessor(md)
+        md.postprocessors.add('checklist', postprocessor, '>raw_html')
+
+
+class ChecklistPostprocessor(Postprocessor):
+    """Markdown extension that supports GitHub styled checklists for tasks, based on https://github.com/FND/markdown-checklist
+
+    For example:
+    - [ ] An incomplete task
+    - [x] A completed task
+    """
+
+    item_with_children_pattern = re.compile(r'<li>\[([ Xx])\]((?!</li>).*)<ul>', re.MULTILINE)
+    item_paragraph_pattern = re.compile(r'<p>\[([ Xx])\](.*)</p>', re.MULTILINE)
+    item_pattern = re.compile(r'<li>\[([ Xx])\](.*)</li>', re.MULTILINE)
```

Comment:
The negative lookahead is clever, though overall you're going to struggle to manipulate HTML and be confident your regexes are bug-free because HTML isn't a regular language.  You need something more expressive like `xml.etree`. 

This would make it more tractable to handle nested lists and my other comments.

Reply:
Absolutely, I'm not proud of the Regex HTML parsing, but it's what was there in the markdown checklist extension that I linked to in the Class docs. `xml.etree` would be a nice improvement, I may look into that at some point.